bundles/org.openhab.binding.hue/src/main/java/org/openhab/binding/hue/internal/discovery/HueBridgeUPNPDiscoveryParticipant.java

```diff
@@ -0,0 +1,162 @@
+/**
+ * Copyright (c) 2010-2023 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.hue.internal.discovery;
+
+import static org.openhab.binding.hue.internal.HueBindingConstants.*;
+
+import java.net.URL;
+import java.util.Dictionary;
+import java.util.Map;
+import java.util.Set;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.jupnp.model.meta.DeviceDetails;
+import org.jupnp.model.meta.ModelDetails;
+import org.jupnp.model.meta.RemoteDevice;
+import org.openhab.binding.hue.internal.handler.HueBridgeHandler;
+import org.openhab.core.config.discovery.DiscoveryResult;
+import org.openhab.core.config.discovery.DiscoveryResultBuilder;
+import org.openhab.core.config.discovery.DiscoveryService;
+import org.openhab.core.config.discovery.upnp.UpnpDiscoveryParticipant;
+import org.openhab.core.config.discovery.upnp.internal.UpnpDiscoveryService;
+import org.openhab.core.thing.Thing;
+import org.openhab.core.thing.ThingTypeUID;
+import org.openhab.core.thing.ThingUID;
+import org.osgi.service.component.ComponentContext;
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Modified;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link HueBridgeUPNPDiscoveryParticipant} is responsible for discovering new and removed Hue Bridges. It uses the
+ * central {@link UpnpDiscoveryService}.
+ *
+ * The discovery through UPnP was replaced by mDNS discovery for recent bridges (V2).
+ * For old bridges (V1), the UPnP discovery is still required (as mDNS is not implemented).
+ * This class allows discovering only old bridges using UPnP.
+ *
+ * @author Laurent Garnier - Initial contribution
+ */
+@Component(configurationPid = "discovery.hue")
+@NonNullByDefault
+public class HueBridgeUPNPDiscoveryParticipant implements UpnpDiscoveryParticipant {
+
+    private static final String EXPECTED_MODEL_NAME_PREFIX = "Philips hue bridge";
+
+    private final Logger logger = LoggerFactory.getLogger(HueBridgeUPNPDiscoveryParticipant.class);
+
+    private long removalGracePeriod = 0L;
```

Comment:
Sorry for being lazy and asking: Is the property required, i.e. can we trust that it is always set? In the old version of the discovery class removed in #11842 the value was 50:
https://github.com/openhab/openhab-addons/pull/11842/files#diff-5cbad99227639ffbe9e3137aeb63db178c2062763f9b711aa7de8bf1ce5ef2a6L58-L59

Reply:
The removal grace period was introduced because sometimes the bridge is too late in sending its SSDP:ALIVE refresh notifications. This has the consequence that the bridge is frequently repeatedly added to and removed from the Inbox. The value that we had set at that time was sufficient to prevent that from occurring.

Reply:
Yes, I remember. Strangely, I was never concerned by this problem.